### PR TITLE
Fix incorrect date handling in Phoenix 4 & 5

### PR DESCRIPTION
**Pull request by ebyhr**

### CODE REVIEW

**Comment by ebyhr** [comment]
CI 
* #7224 https://github.com/trinodb/trino/runs/4916469492
* #5527 https://github.com/trinodb/trino/runs/4916468461

File: plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java [hunk 1]
```diff
@@ -695,6 +704,34 @@ public void renameSchema(ConnectorSession session, String schemaName, String new
         return properties.buildOrThrow();
     }
 
+    private static LongReadFunction dateReadFunction()
+    {
+        return (resultSet, index) -> {
+            // Convert to LocalDate from java.sql.Date via String because java.sql.Date#toLocalDate() returns wrong results in B.C. dates. -5881579-07-11 -> +5881580-07-11
```
Comment (lhofhansl) [inline plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java:710]: Is this a problem in all JDBC based connectors, or specifically in Phoenix?

Comment (ebyhr) [reply plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java:710]: It's not specific to Phoenix.

Comment (lhofhansl) [reply plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java:710]: So the followup question... Is that something to fix in base jdbc, or fix it for each driver?

Comment (ebyhr) [reply plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java:710]: The method is already deprecated. I wouldn't modify in this situation and would fix each connectors.

File: plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java [hunk 2]
```diff
@@ -695,6 +704,34 @@ public void renameSchema(ConnectorSession session, String schemaName, String new
...
+            // Phoenix JDBC driver supports getObject(index, LocalDate.class), but it leads to incorrect issues. -5877641-06-23 -> 7642-06-23 & 5881580-07-11 -> 1580-07-11
+            // Still returns +10 days during julian -> gregorian switch
```
Comment (hashhar) [inline plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java:712]: Unclear. Do you mean that current implementation still returns +10 days during switch or does it refer to `getObject(index, LocalDate.class)`?

**Review by hashhar — APPROVED** [review]
Looks good % question about comment

BTW I notice that Phoenix supports larger DATE range than Trino - do we get reasonable error messages when reading those?

**Comment by ebyhr** [comment]
> Also can we mention "negative dates" in PR title and commit to be more specific?

We shouldn't do this because the issue isn't specific to negative dates.

File: plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java (hunk 2 above)
Comment (ebyhr) [reply plugin/trino-phoenix/src/main/java/io/trino/plugin/phoenix/PhoenixClient.java]: The former. Updated the code comment.

**Pull request merged**